File: Patterns/Iterator Pattern/Cpp/MenuTestDrive.cpp
#include "Menu/PancakeHouseMenu.cpp"
#include "Menu/DinerMenu.cpp"
#include "Waitress.cpp"

int main() {
    PancakeHouseMenu* pancakeHouseMenu = new PancakeHouseMenu();
    DinerMenu* dinerMenu = new DinerMenu();

    Waitress* waitress = new Waitress(pancakeHouseMenu, dinerMenu);

    waitress->printMenu();
}

File: Patterns/Iterator Pattern/CSharp/MenuTestDrive.cs
using Menu;

public class MenuTestDrive
{
    public static void Main(string[] args)
    {
        Menu.PancakeHouseMenu pancakeHouseMenu = new PancakeHouseMenu();
        Menu.DinnerMenu dinnerMenu = new DinnerMenu();

        Waitress waitress = new Waitress(pancakeHouseMenu, dinnerMenu);

        waitress.PrintMenu();
    }
}